conductor-client/src/main/kotlin/com/openlattice/organizations/OrganizationEntitySetsService.kt

```diff
@@ -151,48 +203,89 @@ class OrganizationMetadataEntitySetsService(
             organizationMetadataEntitySetIds.columns
         }
 
+        val schemaEntitySetId = if (organizationMetadataEntitySetIds.schemas == UNINITIALIZED_METADATA_ENTITY_SET_ID) {
+            val schemasEntitySet = buildSchemaEntitySet(organizationId)
+            val id = entitySetsManager.createEntitySet(adminRole.principal, schemasEntitySet)
+            createdEntitySets.add(id)
+            id
+        } else {
+            organizationMetadataEntitySetIds.schemas
+        }
+
+        val viewsEntitySetId = if (organizationMetadataEntitySetIds.views == UNINITIALIZED_METADATA_ENTITY_SET_ID) {
+            val viewsEntitySet = buildViewsEntitySet(organizationId)
+            val id = entitySetsManager.createEntitySet(adminRole.principal, viewsEntitySet)
+            createdEntitySets.add(id)
+            id
+        } else {
+            organizationMetadataEntitySetIds.views
+        }
+
+        val accessRequestsEntitySetId = if (organizationMetadataEntitySetIds.accessRequests == UNINITIALIZED_METADATA_ENTITY_SET_ID) {
+            val accessRequestsEntitySet = buildAccessRequestEntitySet(organizationId)
+            val id = entitySetsManager.createEntitySet(adminRole.principal, accessRequestsEntitySet)
+            createdEntitySets.add(id)
+            id
+        } else {
+            organizationMetadataEntitySetIds.accessRequests
+        }
+
         if (createdEntitySets.isNotEmpty()) {
-            organizationService.setOrganizationMetadataEntitySetIds(
+            setOrganizationMetadataEntitySetIds(
                     organizationId,
                     OrganizationMetadataEntitySetIds(
                             organizationMetadataEntitySetId,
                             datasetsEntitySetId,
-                            columnsEntitySetId
+                            columnsEntitySetId,
+                            schemaEntitySetId,
+                            viewsEntitySetId,
+                            accessRequestsEntitySetId
                     )
             )
 
             entitySetsManager.getEntitySetsAsMap(createdEntitySets).values.forEach {
                 entitySetsManager.setupOrganizationMetadataAndAuditEntitySets(it)
             }
 
-            val orgPrincipal = organizationService.getOrganizationPrincipal(organizationId)!!.principal
+            val orgPrincipal = getOrganizationPrincipal(securePrincipalsManager, organizationId)!!.principal
```

Comment:
seems a bit less mind-bendy to just have some kind of `OrganizationPrincipalsManager` on top of `SecurePrincipalsManager` that implements the logic here (or just adding it to SecurePrincipalsManager) to avoid the whole static + passing a service as an argument + cyclic loophole with two services both calling each other thing, although that can be done in future refactors